pkg/routers/health/error_budget.go

```diff
@@ -0,0 +1,99 @@
+package health
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+const budgetSeparator = "/"
+
+type Unit string
+
+const (
+	MILLI Unit = "ms"
+	MIN   Unit = "m"
+	SEC   Unit = "s"
+	HOUR  Unit = "h"
+)
+
+// ErrorBudget parses human-friendly error budget representation and return it as errors & update rate pair
+// Error budgets could be set as a string in the following format: "10/s", "5/ms", "100/m" "1500/h"
+type ErrorBudget struct {
+	budget int
+	unit   Unit
+}
+
+func NewErrorBudget(budget int, unit Unit) *ErrorBudget {
+	return &ErrorBudget{
+		budget: budget,
+		unit:   unit,
+	}
+}
+
+func DefaultErrorBudget() ErrorBudget {
+	return ErrorBudget{
+		budget: 10,
+		unit:   MIN,
+	}
+}
+
+// Budget defines max allows number of errors per given time period
+func (b *ErrorBudget) Budget() uint64 {
+	return uint64(b.budget)
+}
+
+// TimePerTokenMicro defines how much time do we need to wait to get one error token recovered (in microseconds)
```

Comment:
Is this adjustable?

Reply:
@mkrueger12 yes, it's adjustable and I even connect error budget field to the general config already:
- https://github.com/EinStack/glide/pull/68/files#diff-91b58caeed8701c64ca63ce432a7adfafd02fdc7aca313c113c08c15122cf666
- https://github.com/EinStack/glide/pull/68/files#diff-093f2a2075da7c408d072d5cfbad2cde9608bdc90652f35eaab0100be4cc50eaR20

Users will be able to set up error budget in a form of "10/s" e.g. "tolerate at most 10 errors per second"